src/main/java/org/broadinstitute/hellbender/tools/spark/utils/LongBloomFilter.java

```diff
@@ -0,0 +1,245 @@
+package org.broadinstitute.hellbender.tools.spark.utils;
+
+import com.esotericsoftware.kryo.DefaultSerializer;
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.SVUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.Arrays;
+
+/**
+ * Bloom filter for primitive longs. Useful for quickly querying whether a long is a part of a set when a
+ * a finite false positive probability can be tolerated. Optimal array size and number of hashes are determined
+ * for a given number of elements to be inserted and false prositive probability.
+ */
+@DefaultSerializer(LongBloomFilter.Serializer.class)
+public class LongBloomFilter implements QueryableLongSet {
+
+    // largest prime numbers less than each half power of 2 from 2^8 to 2^31
+    // Note the last size is the greatest prime that is an allowable Java array size (<=2^31-3)
+    private static final int[] legalSizes = {
+            251, 359, 509, 719, 1021, 1447, 2039, 2887, 4093, 5791, 8191, 11579, 16381, 23167, 32749, 46337, 65521,
+            92681, 131071, 185363, 262139, 370723, 524287, 741431, 1048573, 1482907, 2097143, 2965819, 4194301, 5931641,
+            8388593, 11863279, 16777213, 23726561, 33554393, 47453111, 67108859, 94906249, 134217689, 189812507,
+            268435399, 379625047, 536870909, 759250111, 1073741789, 1518500213, 2147483629
+    };
+    private final transient Logger logger = LogManager.getLogger(this.getClass());
+    private final long numBits;
+    private final int numBuckets, numHashes;
+    private final byte[] buckets;
+
+    public LongBloomFilter(final long num_elements, final double fpp) {
+        Utils.validateArg(num_elements > 0, "Number of elements must be greater than 0");
+        Utils.validateArg(fpp > 0 && fpp <= 1, "False positive probability must be between 0 and 1");
+
+        final long optimal_num_bytes = getOptimalNumberOfBytes(num_elements, fpp);
+        this.numBuckets = getLegalSizeAbove(optimal_num_bytes);
+        this.numBits = (this.numBuckets * 8L);
+        final int optimal_num_hashes = (int) Math.round(-Math.log(fpp) / Math.log(2));
+        this.numHashes = optimal_num_hashes > 0 ? optimal_num_hashes : 1;
+        this.buckets = new byte[numBuckets];
+    }
+
+    public LongBloomFilter(final long minNumElements, final int numBytes) {
```

Comment:
Sorry should have removed this beforehand. I don't know what it's doing either, but it isn't used anywhere. I see why it would be better to set numBits to a legal size even. Although this will decrease the max size by a factor of 8, it will not be an issue since I am going to implement your suggestion to maintain a list of bit arrays.